modules/apps/object/object-dynamic-data-mapping-form-field-type/src/main/java/com/liferay/object/dynamic/data/mapping/form/field/type/internal/object/relationship/ObjectRelationshipDDMFormFieldTemplateContextContributor.java

```diff
@@ -235,6 +235,10 @@ private String _getValueKey(DDMFormField ddmFormField) {
 		return systemObjectDefinitionMetadata.getRESTDTOIdPropertyName();
 	}
 
+	private String _replaceDateLabelKey(String date) {
+		return date.replaceAll("createDate", "dateCreated");
+	}
```

Comment:
Regard the logic, as I mentioned before. you can make this method deal with the two situations (when we receive createDate" or "modifiedDate")

To compare strings, Liferay provide a useful class: `StringUtil`
You can use it to validate both situations.

Search for some examples in portal, if you need.